### DISCUSSION TITLE
Propagate span-context for WebSocket

### DISCUSSION THREAD

**Discussion by ychaydhari** [open-telemetry/opentelemetry-java-contrib#493] [Q&A]
Hello,
We have two services running on the machine. Both services communicate with each other through web socket. Can we propagate context through the web socket? or do you have any examples for reference?

Note:- for js one NPM library is available WebSocket context propagation. similarly, is any library for Java?

**Comment by mateuszrzeszutek** [comment]
Hey @ychaydhari ,
As it is now, we're not instrumenting any WebSocket API/library/framework; there are no semantic conventions for WebSocket calls in the OpenTelemetry spec.
If you want to propagate context through WebSockets you'll probably have to do that manually, as described in [the docs](https://opentelemetry.io/docs/instrumentation/java/manual/#context-propagation).
